### ISSUE TITLE
MapView onRegionChange stops working after 5 seconds, and the app crashes 1 min later

### ISSUE THREAD

**Issue by hakosl** [react-community/react-native-maps#174]
I'm trying to get the displaylatLng example to work on android, but it stops updating the state approximately 5 seconds after i start moving the map around, then after ca 1 min the entire app crashes. The issue is very similar to earlier issues with the onRegionChange callback. I'm running using an lg g2 running 5.0.2 on react native 0.22.0
edit: by crashing I mean that the app closes and exits entirely

**Comment by christopherdro** [comment]
@hakosl What does the `onRegionChange` callback do your app?

**Comment by jrichardlai** [comment]
@hakosl do you have a stacktrace that shows the crash?

**Comment by hakosl** [comment]
@christopherdro onRegionChange does this.setState({ region }), which i use to show the current centered location on the map, just like in the example.

**Comment by rt2zz** [comment]
Possibly an infinite loop? I would try putting a debounce on the onRegionChange (e. g. Lodash has a good deboince util)

**Comment by rt2zz** [comment]
Fwiw We have a very similar setup minus the setstate bit, so the issue likely stems from rerender

**Comment by christopherdro** [comment]
It's probably stemming from something else that keeps re-rendering.
You can trying using `onRegionChangeEnd` instead.

Does this crash happen in the example without any modifications?

**Comment by hakosl** [comment]
@christopherdro It's working when i use onRegionChangeEnd, thanks for the help, and sorry for slow replies

**Comment by hakosl** [comment]
On 0.4.2 I am having the same problem again. I have tried solutions mentioned in this thread, but i seem to have no luck with them anymore. @christopherdro Here is my current code, it's pretty similar to the example code.
```
import React, { Component, StyleSheet, View, Text } from 'react-native';

import MapView, { Marker } from 'react-native-maps';

import Meteor from 'react-native-meteor';

import _ from 'lodash';

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  map: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
});

export default class PassengerMap extends Component {
  constructor(props) {
    super(props);

    const position = {
      coords: {
        latitude: null,
        longitude: null,
      },
    };


    this.state = {
      position,
      region: props.initialRegion,
    };
  }

  static defaultProps = {
    drivers: [],
    initialRegion: {
      latitude: 58.144127,
      longitude: 7.983005,
      latitudeDelta: 0.0922,
      longitudeDelta: 0.0421,
    },
  };

  static propTypes = {
    drivers: React.PropTypes.array,
    initialRegion: React.PropTypes.objectOf(React.PropTypes.number),
  };

  componentDidMount() {
    console.log('this.refs.map: ', this.refs.map);

    /* navigator.geolocation.getCurrentPosition(
      this._geolocationUpdate,
      // this._geolocationError,
      // this.geolocationOptions,
    );

    this._geolocationReference = navigator.geolocation.watchPosition(
      this._geolocationUpdate,
      // this._geolocationError,
      // this.geolocationOptions,
    );*/
  }

  componentWillUnmount() {
    navigator.geolocation.clearWatch(this._geolocationReference);
  }

  _geolocationUpdate = pos => {
    const position = JSON.parse(JSON.stringify(pos));
    position.lat = position.coords.latitude;
    position.lng = position.coords.longitude;
    position.latitudeDelta = 0.0922;
    position.longitudeDelta = 0.0421;

    console.log('navigator.geolocation --------');
    console.log('pos', pos);
    console.log('position.coords', position.coords);

    this.setState({ position });

    return position;
    // this.refs.map.animateToCoordinate(position.coords, 1);
  };

  _geolocationError = positionError => {
    this.setState({ positionError });
  };

  _geolocationError = {
    enableHighAccuracy: true,
    timeout: 5000,
    maximumAge: 0,
  };

  _onMapChange = (region) => {
    this.setState({ region });
  };

  render() {
    const { latitude, longitude, latitudeDelta, longitudeDelta } = this.props.initialRegion;

    return (
      <View style={styles.container}>
        <MapView
          ref="map"
          style={ styles.map }
          showsUserLocation
          Region={this.state.region}
          onRegionChange={_.debounce(this._onMapChange, 100)}
        >
          {/* some markers...*/}
        </MapView>
        <Text style={{ marginBottom: 64 }}>
          {`${this.state.region.latitude.toPrecision(7)}, ${this.state.region.longitude.toPrecision(7)}`}
        </Text>
      </View>
    );
  }
}
```

**Comment by christopherdro** [comment]
I made a mistake in my previous comment.

Try `onRegionChangeCompleted`.

**Comment by hakosl** [comment]
@christopherdro i changed `onRegionChange={_.debounce(this._onMapChange, 100)}` to `onRegionChangeComplete={_.debounce(this._onMapChange, 100)}` unfortunatley, I'm getting exactly the same issue as before. 

Regarding crash logs. how do i get these, I tried `adb logcat | grep Google` but i get no output, I've also tried `adb logcat *:S ReactNative:V ReactNativeJS:V`but I'm not getting any useful information after the app has crashed.
